src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.BalanceHandler;
+import core.basesyntax.strategy.impl.OperatorStrategyImpl;
+import core.basesyntax.strategy.impl.PurchaseHandler;
+import core.basesyntax.strategy.impl.ReturnHandler;
+import core.basesyntax.strategy.impl.SupplyHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitShopServiceImplTest {
+    private static FruitShopService fruitShopService;
+    private static Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    @BeforeClass
+    public static void beforeClass() {
+        operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler());
+        Storage.getFruits().clear();
+    }
+
+    @After
+    public void cleanStorage() {
+        Storage.getFruits().clear();
+    }
+
+    @Test
+    public void processTransaction_withValidInformation_Ok() {
```

Comment:
this class should be responsible only for FruitShopServiceImpl testing. So, let's avoid isolated testing of handlers here. I'd recommend adding instead two additional test cases: when list of transaction == null, and when list is empty